### PR TITLE
booleanOperationsExperiments

### PR DESCRIPTION
**Pull request by mauigna06**
create a method to make the surgicalGuidePrototype and improve the GUI

### CODE REVIEW

**Review by mauigna06 — COMMENTED** [review]
I think this code is quite hard to understand. Do you have any suggestions on how to improve it?

File: BoneReconstructionPlanner/BoneReconstructionPlanner.py [hunk 1]
```diff
@@ -781,14 +795,14 @@ def generateFibulaPlanes(self):
         if i!=(len(planeList)-2):
```
Comment (mauigna06) [inline BoneReconstructionPlanner/BoneReconstructionPlanner.py:795]: Starting from here

File: BoneReconstructionPlanner/BoneReconstructionPlanner.py [hunk 2]
```diff
@@ -809,13 +823,16 @@ def generateFibulaPlanes(self):
...
-        deltaz = z0Sup - z1Inf
+        deltaZ = (z0Sup - z0Inf)/2 + (z1Sup - z1Inf)/2
 
-        betweenSpace.append(deltaz)
+        betweenSpace.append(deltaZ)
 
-        self.fibula2FibulaPlanesPositionA.append(self.fibula2FibulaPlanesPositionB[i-1] + fibulaZ*(betweenSpace[i-1]+additionalBetweenSpaceOfFibulaPlanes))
+        self.fibula2FibulaPlanesPositionA.append(self.fibula2FibulaPlanesPositionB[i-1] + fibulaZ*(intersectionDistanceMultiplier*betweenSpace[i-1]+additionalBetweenSpaceOfFibulaPlanes))
 
       self.fibula2FibulaPlanesPositionB.append(self.fibula2FibulaPlanesPositionA[i] + boneSegmentsDistance[i]*fibulaZ)
```
Comment (mauigna06) [inline BoneReconstructionPlanner/BoneReconstructionPlanner.py:838]: Till here. I think this code is hard to understand. Do you have any suggestions on how to improve it?

File: BoneReconstructionPlanner/BoneReconstructionPlanner.py (hunk 1 above)
Comment (lassoan) [inline BoneReconstructionPlanner/BoneReconstructionPlanner.py:795]: Avoid variable names like i and j. They might be acceptable for a loop that contains 1-2 lines, but for anything longer, use a more descriptive name (e.g., planeIndex)

Comment (lassoan) [reply BoneReconstructionPlanner/BoneReconstructionPlanner.py:795]: This whole method is really, really long. A method should not be longer than what fits on 1-2 screens. Try to split it up to smaller methods.

File: BoneReconstructionPlanner/BoneReconstructionPlanner.py (hunk 2 above)
Comment (lassoan) [reply BoneReconstructionPlanner/BoneReconstructionPlanner.py:838]: I think this is quite clear. However, I would recommend to split up this very long method to smaller methods and use the method documentation to explain in 1-2 sentences what the module does.

**Review by lassoan — APPROVED** [review]
The code is quite nice, there is no issue that should prevent merging it into the master. I've added a few suggestions for improvements inline.

**Pull request merged**